src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsServiceTest.kt

```diff
@@ -0,0 +1,67 @@
+package uk.gov.dluhc.printapi.service
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.given
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import uk.gov.dluhc.printapi.database.entity.Status
+import uk.gov.dluhc.printapi.database.repository.PrintDetailsRepository
+import uk.gov.dluhc.printapi.messaging.MessageQueue
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintRequestBatchMessage
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidBatchId
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintDetails
+
+@ExtendWith(MockitoExtension::class)
+class PrintRequestsServiceTest {
+    @InjectMocks
+    private lateinit var printRequestsService: PrintRequestsService
+
+    @Mock
+    private lateinit var idFactory: IdFactory
+
+    @Mock
+    private lateinit var printDetailsRepository: PrintDetailsRepository
+
+    @Mock
+    private lateinit var processPrintRequestQueue: MessageQueue<ProcessPrintRequestBatchMessage>
+
+    @Test
+    fun `should batch print requests, save and submit to queue`() {
+        // Given
+        val batchSize = 5
+        val numOfRequests = 12
+        val items = (1..numOfRequests).map { buildPrintDetails() }
+
+        given(printDetailsRepository.getAllByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)).willReturn(items)
+        given(idFactory.batchId()).willReturn(aValidBatchId(), aValidBatchId(), aValidBatchId(), aValidBatchId())
```

Comment:
4 return values?
Should we set this to return 3 values, and then verify the idFactory was called exactly 3 times?

Reply:
Also, should it return 3 different batch IDs, and that we should assert each updated item has the right batch ID?